Crash: Invalid memory address or nil pointer dereference

### Community Guidelines

- [X] I have read and agree to the [HashiCorp Community Guidelines ](https://www.hashicorp.com/community-guidelines).
- [ ] Vote on this issue by adding a 👍 [reaction](https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue initial description to help the maintainers prioritize.
- [ ] Do not leave "+1" or other comments that do not add relevant information or questions.
- [ ] If you are interested in working on this issue or have submitted a pull request, please leave a comment.

### Terraform

1.10.0

### Terraform Provider

2.10.0

### VMware vSphere

don't know

### Description

Trying to create a machine with a windows_custom section with domain join. The plugin crashed.


### Affected Resources or Data Sources

`vsphere_virtual_machine`

### Terraform Configuration

```hcl
data "vsphere_virtual_machine" "tpt" {
  name          = "template"
  datacenter_id = data.vsphere_datacenter.datacenter.id
}

resource "vsphere_virtual_machine" "vm" {
  name             = "my vm"
  resource_pool_id = data.vsphere_compute_cluster.cluster.resource_pool_id
  datastore_id     = data.vsphere_datastore.datastore.id
  num_cpus         = 2
  memory           = 2048
  guest_id         = data.vsphere_virtual_machine.tpt.guest_id
  scsi_type        = data.vsphere_virtual_machine.tpt.scsi_type
  network_interface {
    network_id   = data.vsphere_virtual_machine.tpt.network_interfaces[0].network_id
    adapter_type = data.vsphere_virtual_machine.tpt.network_interfaces[0].adapter_type
  }
  disk {
    label            = "disk0"
    size             = data.vsphere_virtual_machine.tpt.disks.0.size
    thin_provisioned = data.vsphere_virtual_machine.tpt.disks.0.thin_provisioned
  }
  clone {
    template_uuid = data.vsphere_virtual_machine.tpt.id
    linked_clone = true
    customize {
      network_interface {}
      windows_options {
        computer_name  = "win11"
        join_domain = "domain.name"
        domain_ou = "OU=machines,DC=domain,DC=name"
        domain_admin_user = "root"
        domain_admin_password = "toor"
      }
    }
}
```


### Debug Output

Don't have it

### Panic Output

```shell
data.vsphere_datacenter.datacenter: Reading...
vsphere_virtual_machine.s5fc2bb: Refreshing state... [id=42240905-24c1-70d9-9f4e-6e0deb859099]
 
Planning failed. OpenTofu encountered an error while generating this plan.
 
╷
│ Error: Request cancelled
│
│ The plugin.(GRPCProvider).ReadResource request was cancelled.
╵
╷
│ Error: Request cancelled
│
│ with data.vsphere_datacenter.datacenter,
│ on [main.tf](http://main.tf/) line 28, in data "vsphere_datacenter" "datacenter":
│ 28: data "vsphere_datacenter" "datacenter" {
│
│ The plugin.(GRPCProvider).ReadDataSource request was cancelled.
╵
 
Stack trace from the terraform-provider-vsphere plugin:
 
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0x10933f6]
 
goroutine 97 [running]:
github.com/hashicorp/terraform-provider-vsphere/vsphere.resourceVSphereVirtualMachineRead(0xc000436e80, {0x1290a00, 0xc0000ad560})
/home/runner/work/terraform-provider-vsphere/terraform-provider-vsphere/provider/vsphere/resource_vsphere_virtual_machine.go:469 +0x6d6
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(Resource).read(0x17ec568?, {0x17ec568?, 0xc000373800?}, 0xd?, {0x1290a00?, 0xc0000ad560?})
/home/runner/go/pkg/mod/github.com/hashicorp/terraform-plugin-sdk/v2@v2.34.0/helper/schema/resource.go:811 +0x15f
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(Resource).RefreshWithoutUpgrade(0xc0003ee380, {0x17ec568, 0xc000373800}, 0xc00064f520, {0x1290a00, 0xc0000ad560})
/home/runner/go/pkg/mod/github.com/hashicorp/terraform-plugin-sdk/v2@v2.34.0/helper/schema/resource.go:1117 +0x529
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(GRPCProviderServer).ReadResource(0xc0003d4078, {0x17ec568?, 0xc000373740?}, 0xc00005d880)
/home/runner/go/pkg/mod/github.com/hashicorp/terraform-plugin-sdk/v2@v2.34.0/helper/schema/grpc_provider.go:708 +0x6c5
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(server).ReadResource(0xc0005248c0, {0x17ec568?, 0xc000372c60?}, 0xc0001d60e0)
/home/runner/go/pkg/mod/github.com/hashicorp/terraform-plugin-go@v0.23.0/tfprotov5/tf5server/server.go:783 +0x309
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ReadResource_Handler({0x1589f40, 0xc0005248c0}, {0x17ec568, 0xc000372c60}, 0xc000436480, 0x0)
/home/runner/go/pkg/mod/github.com/hashicorp/terraform-plugin-go@v0.23.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:482 +0x1a6
google.golang.org/grpc.(Server).processUnaryRPC(0xc000227000, {0x17ec568, 0xc000372bd0}, {0x17f1420, 0xc00052e480}, 0xc000406b40, 0xc000418180, 0x22ab248, 0x0)
/home/runner/go/pkg/mod/google.golang.org/grpc@v1.63.2/server.go:1369 +0xdf8
google.golang.org/grpc.(Server).handleStream(0xc000227000, {0x17f1420, 0xc00052e480}, 0xc000406b40)
/home/runner/go/pkg/mod/google.golang.org/grpc@v1.63.2/server.go:1780 +0xe8b
google.golang.org/grpc.(Server).serveStreams.func2.1()
/home/runner/go/pkg/mod/google.golang.org/grpc@v1.63.2/server.go:1019 +0x8b
created by google.golang.org/grpc.(Server).serveStreams.func2 in goroutine 27
/home/runner/go/pkg/mod/google.golang.org/grpc@v1.63.2/server.go:1030 +0x125
 
Error: The terraform-provider-vsphere plugin crashed!
 
This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```

### Expected Behavior

No crash

### Actual Behavior

Panic

### Steps to Reproduce

Don't know

### Environment Details

_No response_

### Screenshots

_No response_

### References

_No response_

--- a/vsphere/resource_vsphere_virtual_machine.go
+++ b/vsphere/resource_vsphere_virtual_machine.go
@@ -467,14 +467,16 @@ func resourceVSphereVirtualMachineRead(d *schema.ResourceData, meta interface{})
 	if vprops.ResourcePool != nil {
 		_ = d.Set("resource_pool_id", vprops.ResourcePool.Value)
 	}
+
 	// If the VM is part of a vApp, InventoryPath will point to a host path
 	// rather than a VM path, so this step must be skipped.
 	var vmContainer string
 	if vprops.ParentVApp != nil {
 		vmContainer = vprops.ParentVApp.Value
-	} else {
+	} else if vprops.ResourcePool != nil {
 		vmContainer = vprops.ResourcePool.Value
 	}
+
 	if !vappcontainer.IsVApp(client, vmContainer) {
 		f, err := folder.RootPathParticleVM.SplitRelativeFolder(vm.InventoryPath)
 		if err != nil {

